aptos-move/move-examples/data_structures/sources/smart_vector.move

```diff
@@ -0,0 +1,461 @@
+    /// This is O(1), but does not preserve ordering of elements in the vector.
+    /// Aborts if `i` is out of bounds.
+    public fun swap_remove<T>(v: &mut SmartVector<T>, i: u64): T {
+        let len = length(v);
+        assert!(i < len, error::invalid_argument(EINDEX_OUT_OF_BOUNDS));
+        let inline_len = vector::length(&v.inline_vec);
+        let big_vec_wrapper = &mut v.big_vec;
+        let inline_vec = &mut v.inline_vec;
+        if (i >= inline_len) {
+            let big_vec = vector::pop_back(big_vec_wrapper);
+            let val = big_vector::swap_remove(&mut big_vec, i - inline_len);
+            if (big_vector::is_empty(&big_vec)) {
+                big_vector::destroy_empty(big_vec)
+            } else {
+                vector::push_back(big_vec_wrapper, big_vec);
+            };
+            val
+        } else {
+            let val = vector::swap_remove(inline_vec, i);
+            if (inline_len < len) {
+                let big_vec = vector::pop_back(big_vec_wrapper);
+                let last_from_big_vec = big_vector::pop_back(&mut big_vec);
+                if (big_vector::is_empty(&big_vec)) {
+                    big_vector::destroy_empty(big_vec)
+                } else {
+                    vector::push_back(big_vec_wrapper, big_vec);
+                };
+                vector::push_back(inline_vec, last_from_big_vec);
+                vector::swap(inline_vec, i, inline_len - 1);
+            };
+            val
+        }
+    }
+
+    /// Swap the elements at the i'th and j'th indices in the vector v. Will abort if either of i or j are out of bounds
+    /// for v.
+    public fun swap<T: store>(v: &mut SmartVector<T>, i: u64, j: u64) {
+        if (i > j) {
+            return swap(v, j, i)
+        };
+        let len = length(v);
+        assert!(j < len, error::invalid_argument(EINDEX_OUT_OF_BOUNDS));
+        let inline_len = vector::length(&v.inline_vec);
+        if (i >= inline_len) {
+            big_vector::swap(vector::borrow_mut(&mut v.big_vec, 0), i - inline_len, j - inline_len);
+        } else if (j < inline_len) {
+            vector::swap(&mut v.inline_vec, i, j);
+        } else {
+            let big_vec = vector::borrow_mut(&mut v.big_vec, 0);
+            let small_vec = &mut v.inline_vec;
+            let element_i = vector::swap_remove(small_vec, i);
+            let element_j = big_vector::swap_remove(big_vec, j - inline_len);
+            vector::push_back(small_vec, element_j);
+            vector::swap(small_vec, i, inline_len - 1);
+            big_vector::push_back(big_vec, element_i);
+            big_vector::swap(big_vec, j - inline_len, len - inline_len - 1);
+        }
+    }
+
+    /// Reverse the order of the elements in the vector v in-place.
+    /// disclaimer: this function may be costly. Use it at your own discretion.
+    public fun reverse<T: store>(v: &mut SmartVector<T>) {
```

Comment:
similar comment with big_vec: can leverage native vector::swap to be less costly / faster